### PR TITLE
sonar: add onboarding instructions

### PR DESCRIPTION
**Pull request by ray-harris**
.travis.yml:
- Add dist key
- Add addons key
- Fix name for build stage
- Fix name for unit-test stage
- Add SonarCloud commands to unit-test stage

README.md:
- Add SonarCloud onboarding instructions

sonar-project.properties:
- Add generic template since the language for this repo is python

sonar-project.properties.go.example:
- Add template for a Go project

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,12 @@ python: "3.7"
 services:
   - docker
 
+addons:
+  sonarcloud:
+    organization: "open-cluster-management"
+    token:
+      secure: "VlhgSso+hJhaJP/mQVOh8pqD/hcSrkyn187G3bZXKI2uF/dNY8zV1Opy0WJRiEoGJ8AXr9N7Hx6lkVxMcpSd6DNwL2SGc0Rshy/EVcG7jnGZVDo/lsKGrC9lZEWkT9t/ZCij68P9YncWObtvrjsPn2575/qM8yAI05hjAhpPdK80g0+qzqjCGW595i0mEygvTAmcLbRHPJQZ0MAIt0WNsWkEVfBlilgDnUNT5axbnbGNWexBr6JnuRPeSIN5cF1TEyQfZvlSzwQxZLQ4/fDHGez2MwFIY3otoV2J0NVr0SKSiFNbNcJo7aVx8oIB4HtSPCUgvB6TaociCa/uX9au6pPuwSX8vCOGY0OB+1M9m4L1VGHmc9O8yI6437aOCccTxYeorUz44SImzBv0lq5XNHHqHebnDyeYXyTjTnn9Uqly5pC+2DsXCx8yIez+htizu9AMHEpaxLJtom//Lmd+gwkLZhtY45dRkKr8e4cgZIgQpKaq6YrmSPyquuYTUBzPDAaLW4xtXwTktKOBMoTTGp1Q46mrL7vHcmrtR3a8DGpc7Hc4RSSZx0O3nAiUXYaZBdAtNo6NTAYF+uN2bWMxcan8akXvJ6PSF5HdLELsEB9aZEZ7rcIdeQ2BOov/+vjI9M1WFQsp6A4KRN9+NVil5wKlv2magn5TCkKTVVLuzk4="
```
Comment (gurnben) [inline .travis.yml]: Missed this the first time, what is this token?

Comment (ray-harris) [reply .travis.yml]: If you read the README.md section on how to onboard a repo to SonarCloud, you'll see that it's a secure token generated by running `travis encrypt` on the the token give to you by SonarCloud for the new repo.

Comment (gurnben) [reply .travis.yml]: Understood, is there any risk to including this in a public repo?

Comment (ray-harris) [reply .travis.yml]: No. https://docs.travis-ci.com/user/encryption-keys/

Comment (gurnben) [reply .travis.yml]: Ok, good to go then!

**Pull request merged**